.github/workflows/build-test.yml

```diff
@@ -76,7 +76,7 @@ jobs:
     needs: pre-flight-check
     steps:
       - name: Setup ROS
-        uses: ros-tooling/setup-ros@v0.3
+        uses: ros-tooling/setup-ros@v0.5
         with:
           required-ros-distributions: foxy
```

Comment:
What is not considered is that we need to install manully additional dependencies (e.g. pip packages like colcon) via `rosdep`. In this case I would try to avoid that manual work here and rely on the GitHub action proved by the ROS community for the setup of the ROS environment.

Reply:
A few years ago I used those repos but I cannot recall anymore if I had to use pip for some packages.